test/unit/mock/fake_runtime_client.py

```diff
@@ -117,6 +117,7 @@ def __init__(
         """Initialize a fake job."""
         self._job_id = job_id
         self._status = final_status or "QUEUED"
+        self._state = {"status": final_status or "QUEUED"}
```

Comment:
This appears redundant. The API needs this redundancy to not break existing clients. In our own mocks I think we should no longer include `self._status`.  This gives us more confidence that we no longer depend on a soon-to-be-removed response property.